StoreTestWPF.ViewModel/ViewModels/StoreViewModel.cs

```diff
@@ -0,0 +1,160 @@
+                    this.OnPropertyChanged(nameof(this.SelectedCake));
+
+                    this.isCakeSelected = !(value == null);
+                    this.OnPropertyChanged(nameof(this.IsCakeSelected));
+                }
+            }
+        }
+
+        public bool IsCakeSelected
+        {
+            get
+            {
+                return isCakeSelected;
+            }
+            set
+            {
+                if (this.isCakeSelected != value)
+                {
+                    this.isCakeSelected = value;
+                    this.OnPropertyChanged(nameof(this.IsCakeSelected));
+                }
+            }
+        }
+
+        public ICommand LoadCommand => this.loadCommand ?? new RelayCommand(LoadExecuted, LoadCanExecute);
+
+        public ICommand AddCommand => this.addCommand ?? new RelayCommand(AddExecuted);
+
+        public ICommand EditCommand => this.editCommand ?? new RelayCommandWithParameter<CakeViewModel>(EditExecuted, EditCanExecute);
+
+        public ICommand DeleteCommand => this.deleteCommand ?? new RelayCommandWithParameter<CakeViewModel>(DeleteExecuted, DeleteCanExecute);
+
+        public StoreViewModel(IViewService viewService)
+        {
+            this.Cakes = new ObservableCollection<CakeViewModel>();
+            this.dbContext = CakeStoreDbContext.Create();
+            this.viewService = viewService;
+        }
+
+        private void AddExecuted()
+        {
+            try
+            {
+                var modifyViewModel = new ModifyCakeViewModel(this.viewService);
+                modifyViewModel.ModifiedCake = new CakeViewModel(new Cake());
+                WindowTitles Title = WindowTitles.Add;
+                modifyViewModel.Title = Title.GetDisplayAttributesFrom(typeof(WindowTitles)).Name;
+
+                if (!this.viewService.ShowWindow(modifyViewModel))
+                {
+                    return;
+                }
+                this.SelectedCake = modifyViewModel.ModifiedCake;
+                this.dbContext.Cakes.Add(SelectedCake.Cake);
+                this.dbContext.SaveChanges();
+
+                this.Cakes.Add(SelectedCake);
+            }
+            catch (Exception exception)
+            {
+                viewService.ShowErrorMessage(exception.Message);
+            }
+            
+        }
+
+        private void EditExecuted(CakeViewModel cake)
+        {
+            try
+            {
+                var unmodifiedCake = new CakeViewModel(this.SelectedCake.Cake.Clone() as Cake);
+                var modifyViewModel = new ModifyCakeViewModel(this.viewService);
+                modifyViewModel.ModifiedCake = cake;
+
+                WindowTitles Title = WindowTitles.Edit;
+                modifyViewModel.Title = Title.GetDisplayAttributesFrom(typeof(WindowTitles)).Name;
+
+                if (!this.viewService.ShowWindow(modifyViewModel))
+                {
+                    this.Cakes[this.Cakes.IndexOf(cake)] = unmodifiedCake;
```

Comment:
What if there is no such cake in collection?

Reply:
Fixed